ngraph/core/src/op/random_uniform.cpp

```diff
@@ -0,0 +1,147 @@
+    if (input_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              input_shape.rank() == 1,
+                              "The rank of the tensor defining output shape must be equal to 1.");
+        if (const auto& const_shape = get_constant_from_source(input_value(0)))
+        {
+            output_shape = PartialShape(const_shape->cast_vector<int64_t>());
+        }
+    }
+
+    const auto& min_pshape = get_input_partial_shape(1);
+    const auto& max_pshape = get_input_partial_shape(2);
+    if (min_pshape.is_static())
+    {
+        const auto& min_rank = min_pshape.rank().get_length();
+        NODE_VALIDATION_CHECK(this, min_rank <= 1, "Min value must be a scalar or 1D tensor.");
+
+        if (min_rank == 1)
+        {
+            NODE_VALIDATION_CHECK(
+                this, min_pshape.compatible(Shape{1}), "'min_val' should have 1 element.");
+        }
+    }
+
+    if (max_pshape.is_static())
+    {
+        const auto& max_rank = max_pshape.rank().get_length();
+        NODE_VALIDATION_CHECK(this, max_rank <= 1, "Max value must be a scalar or 1D tensor.");
+
+        if (max_rank == 1)
+        {
+            NODE_VALIDATION_CHECK(
+                this, max_pshape.compatible(Shape{1}), "'max_val' should have 1 element.");
+        }
+    }
+
+    const element::Type& min_element_type = get_input_element_type(1);
+    element::Type max_element_type = get_input_element_type(2);
+    NODE_VALIDATION_CHECK(this,
+                          min_element_type == max_element_type,
+                          "'min_val' should have the same type as 'max_val'.");
+    NODE_VALIDATION_CHECK(
+        this,
+        min_element_type == get_out_type(),
+        "'min_val' and 'max_val' should have the same type as 'out_type' attribute.");
+
+    if (const auto& const_min = get_constant_from_source(input_value(1)))
+    {
+        if (const auto& const_max = get_constant_from_source(input_value(2)))
+        {
+            if (get_out_type() == ngraph::element::Type_t::i64 ||
+                get_out_type() == ngraph::element::Type_t::i32)
+            {
+                int64_t min_val = const_min->cast_vector<int64_t>()[0];
+                int64_t max_val = const_max->cast_vector<int64_t>()[0];
+
+                NODE_VALIDATION_CHECK(this,
+                                      min_val < max_val,
+                                      "Min value must be less than max value. Got "
+                                      "min value: ",
+                                      min_val,
+                                      ", max value: ",
+                                      max_val);
+            }
+            else if (get_out_type() == ngraph::element::Type_t::f16 ||
+                     get_out_type() == ngraph::element::Type_t::f32 ||
+                     get_out_type() == ngraph::element::Type_t::f64 ||
+                     get_out_type() == ngraph::element::Type_t::bf16)
```

Comment:
If I remember right, we can use `is_real()` here. I propose to do it separately.

Reply:
Done.